homeassistant/components/homekit/type_fans.py

```diff
@@ -0,0 +1,159 @@
+"""Class to hold all light accessories."""
+import logging
+
+from pyhap.const import CATEGORY_FAN
+
+from homeassistant.components.fan import (
+    ATTR_DIRECTION, ATTR_OSCILLATING, ATTR_SPEED, ATTR_SPEED_LIST,
+    DIRECTION_FORWARD, DIRECTION_REVERSE, SUPPORT_DIRECTION,
+    SUPPORT_OSCILLATE, SUPPORT_SET_SPEED)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_SUPPORTED_FEATURES, STATE_OFF, STATE_ON,
+    SERVICE_TURN_OFF, SERVICE_TURN_ON)
+from homeassistant.core import split_entity_id
+
+from . import TYPES
+from .accessories import HomeAccessory, debounce
+from .const import (
+    CHAR_ACTIVE, CHAR_ROTATION_DIRECTION, CHAR_ROTATION_SPEED,
+    CHAR_SWING_MODE, SERV_FANV2)
+
+_LOGGER = logging.getLogger(__name__)
+
+HASS_TO_HOMEKIT = {DIRECTION_FORWARD: 0,
+                   DIRECTION_REVERSE: 1}
+HOMEKIT_TO_HASS = {c: s for s, c in HASS_TO_HOMEKIT.items()}
+
+
+@TYPES.register('Fan')
+class Fan(HomeAccessory):
+    """Generate a Fan accessory for a fan entity.
+
+    Currently supports: state, speed, oscillate, direction.
+    """
+
+    def __init__(self, *args, config):
+        """Initialize a new Light accessory object."""
+        super().__init__(*args, category=CATEGORY_FAN)
+        self._domain = split_entity_id(self.entity_id)[0]
+        self._flag = {CHAR_ACTIVE: False,
+                      CHAR_ROTATION_SPEED: False,
+                      CHAR_ROTATION_DIRECTION: False,
+                      CHAR_SWING_MODE: False}
+        self._state = 0
+
+        self.chars = []
+        self._features = self.hass.states.get(self.entity_id) \
+            .attributes.get(ATTR_SUPPORTED_FEATURES)
+        if self._features & SUPPORT_SET_SPEED:
+            self.chars.append(CHAR_ROTATION_SPEED)
+            self._speed = None
+        if self._features & SUPPORT_DIRECTION:
+            self.chars.append(CHAR_ROTATION_DIRECTION)
+            self._direction = None
+        if self._features & SUPPORT_OSCILLATE:
+            self.chars.append(CHAR_SWING_MODE)
+            self._oscillating = None
+
+        serv_fan = self.add_preload_service(SERV_FANV2, self.chars)
+        self.char_active = serv_fan.configure_char(
+            CHAR_ACTIVE, value=self._state, setter_callback=self.set_state)
+
+        if CHAR_ROTATION_SPEED in self.chars:
+            speed_list = self.hass.states.get(self.entity_id) \
```

Comment:
Does this change over time? At the moment no homekit type accounts for chnages in supported features, so I think changing this to `self.speed_list` would work.

Reply:
No, you're correct in that it doesn't change after being initialized at startup.